core/navigation.js

```diff
@@ -439,8 +467,15 @@ Blockly.Navigation.keyboardOut = function() {
  * Mark the current location of the cursor.
  */
 Blockly.Navigation.markConnection = function() {
-  Blockly.Navigation.insertionNode_ =
-    Blockly.Navigation.cursor_.getCurNode();
+  var curNode = Blockly.Navigation.cursor_.getCurNode();
+  var location = curNode.getLocation();
+
+  //TODO: Add a way to identify if something is connection, field, block, ws
```

Comment:
Isn't that what the node type is for?

Reply:
Yeah except the only way to identify something is a connection is to check if it is either next,previous,output, or input. Soooooo I think what I actually want is just a helper function 'isConnection'.